### PR TITLE
Fix --shell option for adduser

### PR DESCRIPTION
**Pull request by EmilySeville7cfg**
## Description

Hide all comments.

### CODE REVIEW

**Comment by IlanCosman** [comment]
Just to be sure: have you tried using `string` commands instead of `sed`? While there are sprinklings of `sed` here and there, we'd prefer not to add any more of them, and remove the existing ones.

**Comment by EmilySeville7cfg** [comment]
Thanks for a comment! I've removed sed usage in my fix.

File: share/completions/adduser.fish [hunk 1]
```diff
@@ -15,7 +15,7 @@ complete -c adduser -l gid -d 'When creating a group, force the groupid to be th
 complete -c adduser -l group -d 'Create a group'
 complete -c adduser -l help -d 'Display brief instructions'
 complete -c adduser -l home -d 'Use specified directory as the user\'s home directory' -x -a '(__fish_complete_directories)'
-complete -c adduser -l shell -d 'Use shell as the user\'s login shell, rather than the default' -x -a '(cat /etc/shells)'
+complete -c adduser -l shell -d 'Use shell as the user\'s login shell, rather than the default' -x -a '(string match --regex --invert "^\s*#" < /etc/shells)'
```
Comment (zanchey) [inline share/completions/adduser.fish:18]: Nice one. The `chsh` completions use `string match -r '^[^#].*'`, which I think is slightly more correct (based on FreeBSD's documentation and the source for Debian's `shadow`).

Comment (EmilySeville7cfg) [reply share/completions/adduser.fish]: Thanks. ;)

**Pull request merged**